### PR TITLE
Update How Shipping Options & Confirmation Types are defined

### PR DESCRIPTION
**Pull request by justin-shipstation**
One of the major issues I have seen contractors have is they don't understand that confirmation types doesn't work like a grab bag, it is basically just overwriting a known confirmation type and you can only have 1 to 1. This was also frustrating for the magic strings present in Shipping Options, now there is an enum they can use and it also works like a dictionary on the definition side.

Changing app definitions from this
``` javascript
ShippingOptions: [
    {
      Name: 'Contains Dry Ice',
      Type: 'contains-dry-ice',
    },
    {
      Name: 'Contains Alcohol',
      Type: 'contains-alcohol',
    }
  ]
```
to this
``` javascript
ShippingOptions: {
    [ShippingOptionEnum.DryIce]: 'Contains Dry Ice',
    [ShippingOptionEnum.ContainsAlcohol]: undefined, // default name will be used
  },
```

And changing this
``` javascript
DefaultConfirmationTypes: [
    {
        Name: 'No Confirmation Required',
        Type: ConfirmationTypeType.None
    },
    {
        Name: 'Adult Required',
        Type: ConfirmationTypeType.AdultSignature
    }
]
```

to this
``` javascript
DefaultConfirmationTypes: {
    [ConfirmationTypeEnum.None]: 'No Confirmation Required',
    [ConfirmationTypeEnum.AdultSignature]: 'Adult Required',
  },
```

### CODE REVIEW

File: packages/carrier-api/src/app/metadata/shipping-option.ts [hunk 1]
```diff
@@ -1,4 +1,34 @@
 export interface ShippingOption {
   Name: string;
-  Type: string;
+  Type: ShippingOptionEnum;
+}
+export type ShippingOptionDictionary = { [Key in ShippingOptionEnum]?: string };
+
+export enum ShippingOptionEnum {
```
Comment (dougwile) [inline packages/carrier-api/src/app/metadata/shipping-option.ts:7]: What is the source of these enum values?

Comment (justin-shipstation) [reply packages/carrier-api/src/app/metadata/shipping-option.ts:7]: @dougwile https://github.com/shipstation/throatpunch/blob/master/client/services/other-shipping-options/constants.js

These are the constants the front end use from the registry to drive the front end.

Comment (jbolton209) [reply packages/carrier-api/src/app/metadata/shipping-option.ts:7]: Feels like we should eventually consolidate these strings into a single place across SS apps/services, we now have this same file 3 places in our codebase. Once V2 goes away there will be 2 places (V3 & here, unless there are more places I dont know about).

**Review by jbolton209 — APPROVED** [review]

**Comment by justin-shipstation** [comment]
increment major

**Pull request merged**